Cython/Compiler/ExprNodes.py

```diff
@@ -3119,19 +3119,18 @@ def generate_next_sequence_item(self, test_name, result_name, code):
             inc_dec = '--'
         else:
             inc_dec = '++'
+        if test_name == 'List':
+            getitem = "__Pyx_PyList_GetItemRef"
+            incref = "__Pyx_GOTREF"
+        else:  # Tuple
+            getitem = "PyTuple_GET_ITEM"
+            incref = "__Pyx_INCREF"
         code.putln("#if CYTHON_ASSUME_SAFE_MACROS && !CYTHON_AVOID_BORROWED_REFS")
         code.putln(
-            "%s = Py%s_GET_ITEM(%s, %s); __Pyx_INCREF(%s); %s%s; %s" % (
-                result_name,
-                test_name,
-                self.py_result(),
-                self.counter_cname,
-                result_name,
-                self.counter_cname,
-                inc_dec,
-                # use the error label to avoid C compiler warnings if we only use it below
-                code.error_goto_if_neg('0', self.pos)
-                ))
+            f"{result_name} = {getitem}({self.py_result()}, {self.counter_cname}); "
+            f"{incref}({result_name}); "
+            f"{self.counter_cname}{inc_dec}; "
+            f"{code.error_goto_if_neg('0', self.pos)}")
```

Comment:
Let's keep the comment.
```suggestion
            # Use the error label to avoid C compiler warnings if we only use it below.
            f"{code.error_goto_if_neg('0', self.pos)}")
```